src/plugins/core/pivot.ts

```diff
@@ -228,27 +246,73 @@ export class PivotCorePlugin extends CorePlugin<CoreState> implements CoreState
     formulaId = this.nextFormulaId.toString()
   ) {
     this.history.update("pivots", pivotId, { definition: deepCopy(pivot), formulaId });
-    this.compileCalculatedMeasures(pivot.measures);
+    this.compileCalculatedMeasures(pivotId, pivot.measures);
     this.history.update("formulaIds", formulaId, pivotId);
     this.history.update("nextFormulaId", this.nextFormulaId + 1);
   }
 
-  private compileCalculatedMeasures(measures: PivotCoreMeasure[]) {
+  private compileCalculatedMeasures(pivotId: UID, measures: PivotCoreMeasure[]) {
     for (const measure of measures) {
       if (measure.computedBy) {
-        const sheetId = measure.computedBy.sheetId;
         const compiledFormula = this.compileMeasureFormula(
           measure.computedBy.sheetId,
           measure.computedBy.formula
         );
         this.history.update(
           "compiledMeasureFormulas",
-          sheetId,
-          measure.computedBy.formula,
+          pivotId,
+          measure.id,
+          "formula",
           compiledFormula
         );
       }
     }
+    for (const measure of measures) {
+      if (measure.computedBy) {
+        const dependencies = this.computeMeasureFullDependencies(pivotId, measure);
+        this.history.update(
```

Comment:
Can probably be done in the same loop as the complied formula, if `computeMeasureFullDependencies` returned the whole `MeasureState` object.